Tutorial.md

```diff
@@ -11,477 +12,247 @@ Gist is a paste. A snippet of text. Not unrelated to [GitHub Gists][].  To set s
+### URI Template
 
-    ### Create a Gist [POST]
-    To create a new Gist simply provide a JSON hash of the *description* and *content* attributes for the new Gist.
-
-    + Request (application/json)
-
-            {
-                "description": "Description of Gist",
-                "content": "String content"
-            }
-
-    + Response 201 (application/hal+json)
-
-        [Gist][]
-
-At this point you should be familiar with everything in this listing except the definition and discussion of the query parameter *"since"*:
-
-<a name="query_parameters"></a>
-#### Query Parameters
-
-    ## Gists Collection [/gists{?since}]
-
-As with the URI parameters the Query parameters are defined in the [URI Template][] format. The difference here is that a query parameter name is preceded by the *question mark* and it is always at the end of the URI Template.
-
-> **NOTE:** To define multiple query parameters simply comma-separate their names. For example `{?month,year}`.
-
-#### Action-specific Query Parameters
-
-    ### List All Gists [GET]
-    + Parameters
-        + since (optional, string) ... Timestamp in ISO 8601 format: `YYYY-MM-DDTHH:MM:SSZ` Only Gists updated at or after this time are returned.
-
-Often a resource query parameter is specific to just one of the resource actions. In this case you can discuss it in the relevant action only. No matter where you discuss a query parameter it follows exactly the same syntax as with [URI Parameters][].
-
-## Subsequent Resource
-The last missing piece in the expected functionality is the Star Resource and its actions. Utilizing everything we have learned so far the Star Resource and its actions are defined as follows:
-
-    ## Star [/gists/{id}/star]
-    Star resource represents a Gist starred status.
-
-    The Star resource has the following attribute:
-
-    - starred
+The URI for the “Question” resource uses a variable component, expressed by [URI Template][]. In this case, we have an id variable called `{question_id}. Where `question_id` is substituted for a real `id`.
 
+<a id="uri-parameters"></a>
+### URI Parameters
 
-    + Parameters
+URI parameters can be described in the URI using a list of Parameters. For “Question” it would be as follows:
 
-        + id (string) ... ID of the gist in the form of a hash
+```markdown
++ Parameters
+    + question_id (number) ... ID of the Question in form of an integer
+```
 
-    + Model (application/hal+json)
+The "`question_id` variable of the URI template is a parameter for every action on this resource. Here defined of an arbitrary type `number`, followed by a description for the parameter.
 
-        HAL+JSON representation of Star Resource.
+> Refer to API Blueprint Specification's [Resource Parameters Section][] for additional examples.
 
-        + Headers
+### Actions
 
-                Link: <http:/api.gistfox.com/gists/42/star>;rel="self"
+Similar to before, this resource has an action to view the question detail.
```

Comment:
At this point – reading really slowly and trying to get the grasp on what is API Blueprint – I do not recall what you are referring to in "similar to before" (remove?)